### PR TITLE
[nearlib] add wait for transaction to complete to integration test

### PR DESCRIPTION
**Pull request by janedegtiareva**
this helps reduce test flakiness

### CODE REVIEW

File: nearlib/account.js [hunk 1]
```diff
@@ -27,9 +27,8 @@ class Account {
         const keyWithRandomSeed = await KeyPair.fromRandomSeed();
         const createAccountResult = await this.createAccount(
             newAccountId, keyWithRandomSeed.getPublicKey(), amount, originatorAccountId);
-        const response = {};
-        response["key"] = keyWithRandomSeed;
-        return response;
+        createAccountResult["key"] = keyWithRandomSeed;
+        return createAccountResult;
```
Comment (vgrichina) [inline nearlib/account.js]:
```js
return { key: keyWithRandomSeed, ...createAccountResult }
```
would be cleaner JS

File: nearlib/test/integration.test.js [hunk 2]
```diff
@@ -159,13 +156,11 @@ const waitForContractToDeploy = async (deployResult) => {
...
 };
 
-const waitForNonceToIncrease = async (initialAccount) => {
-    await callUntilConditionIsMet(
-        async () => { return await account.viewAccount(initialAccount['account_id']); },
-        (response) => { return response['nonce'] != initialAccount['nonce'] },
-        "Call view account until nonce increases"
-    );
-};
+function sleep(time) {
```
Comment (vgrichina) [inline nearlib/test/integration.test.js]: alternatively this also works in Node.js:
```js
const sleep = require('utils').promisify(setTimeout);
```
this approach should generally work for various functions taking callbacks

File: nearlib/test/integration.test.js [hunk 3]
```diff
@@ -147,9 +124,29 @@ const callUntilConditionIsMet = async (functToPoll, condition, description, maxR
...
+    fail('exceeded number of retries for ' + description);
 };
 
+const waitForTransactionToComplete = async (submitTransactionResult) => {
+    expect(submitTransactionResult.hash).not.toBeFalsy();
+    console.log("Waiting for transaction" + submitTransactionResult.hash);
+    await callUntilConditionIsMet(
+        async () => { return await nearClient.getTransactionStatus(submitTransactionResult.hash); },
+        (response) => {
+            if (response['status'] == 'Completed') {
```
Comment (vgrichina) [inline nearlib/test/integration.test.js]: `response.status`

File: nearlib/test/integration.test.js [hunk 4]
```diff
@@ -147,9 +124,29 @@ const callUntilConditionIsMet = async (functToPoll, condition, description, maxR
...
     }
+    fail('exceeded number of retries for ' + description);
 };
 
+const waitForTransactionToComplete = async (submitTransactionResult) => {
+    expect(submitTransactionResult.hash).not.toBeFalsy();
+    console.log("Waiting for transaction" + submitTransactionResult.hash);
+    await callUntilConditionIsMet(
+        async () => { return await nearClient.getTransactionStatus(submitTransactionResult.hash); },
+        (response) => {
```
Comment (vgrichina) [inline nearlib/test/integration.test.js]: I would write this whole block as:
`response => response.status == 'Completed`

Comment (janedegtiareva) [reply nearlib/test/integration.test.js]: I want the log though

File: nearlib/test/integration.test.js (hunk 3 above)
Comment (janedegtiareva) [reply nearlib/test/integration.test.js]: done

File: nearlib/account.js (hunk 1 above)
Comment (janedegtiareva) [reply nearlib/account.js]: done

**Pull request merged**